### ISSUE TITLE
Terragrunt always reinitializes state if TF_PLUGIN_CACHE_DIR is specified

### ISSUE THREAD

**Issue by misha-tectonic** [gruntwork-io/terragrunt#839]
Terragrunt does not know that plugins may be stored in a directory specified in `TF_PLUGIN_CACHE_DIR` directory, and it causes it to call `terraform init` every time.

How to reproduce:
```
TF_PLUGIN_CACHE_DIR=somewhere terragrunt plan
TF_PLUGIN_CACHE_DIR=somewhere terragrunt plan
```

**Comment by misha-tectonic** [comment]
I have a patch fixing this issue. I'd appreciate a guidance how to create a test though.

**Comment by brikis98** [comment]
Ah, good point. Thx for catching this.

Testing this is indeed tricky, for if you set env vars, it affects the whole test environment, and could break other tests running concurrently. Is there any other way to configure these settings locally?

**Comment by misha-tectonic** [comment]
Can be set in configuration file: https://www.terraform.io/docs/commands/cli-config.html (so that both ways should be tested)

However, `TF_DATA_DIR` from #838 cannot: https://github.com/hashicorp/terraform/blob/0f2c4b3b68d8ec5b2273c23e478feed34cdf7176/commands.go#L54

**Comment by misha-tectonic** [comment]
Why would an envvar break other tests? It needs to be set only for a single `terragrunt` invocation (in `runTerragruntCommand` AFAIU).

**Comment by misha-tectonic** [comment]
Ah, `runTerragruntCommand` does not run the binary, it runs an entrypoint function inside the process?

**Comment by brikis98** [comment]
Env vars are global for a process and all the tests run in parallel, so this setting would affect all the tests running at the same time. Looks like that configuration file is global too...

Perhaps we can test this by setting those env vars in an [extra_arguments block](https://github.com/gruntwork-io/terragrunt#motivation-2)?

**Comment by brikis98** [comment]
> Ah, runTerragruntCommand does not run the binary, it runs an entrypoint function inside the process?

Yup

**Comment by misha-tectonic** [comment]
After some testing I have understood that `TF_PLUGIN_CACHE_DIR` is not something Terragrunt should care about: all it needs to check is binaries inside state directory, and that is already covered by #838.
